src/main/java/za/co/neslotech/spring/trainspringbootinsight/controller/UserController.java

```diff
@@ -0,0 +1,31 @@
+package za.co.neslotech.spring.trainspringbootinsight.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import za.co.neslotech.spring.trainspringbootinsight.model.User;
+import za.co.neslotech.spring.trainspringbootinsight.repository.IUserRepository;
+
+@RestController
+@RequestMapping("/api/v1/users")
+public class UserController extends AGenericController<User, Long>{
+
+    @Autowired
+    private IUserRepository userRepository;
```

Comment:
Following the pattern of marking everything final that can be final, I would mark this as a final variable and then initialise it in the constructor. The Autowiring will occur automatically